File: extra5/app/src/main/java/com/example/extra5/db/NewsDb.java
package com.example.extra5.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {NewsEntity.class}, version = 1)
public abstract class NewsDb extends RoomDatabase {
    private static volatile NewsDb I;
    public abstract NewsDao dao();

    public static NewsDb get(Context c) {
        if (I == null) {
            synchronized (NewsDb.class) {
                if (I == null) {
                    I = Room.databaseBuilder(c.getApplicationContext(), NewsDb.class, "news.db")
                            .allowMainThreadQueries()
                            .build();
                }
            }
        }
        return I;
    }
}

File: extra5/app/src/main/java/com/example/extra5/db/NewsDb.kt
package com.example.extra5.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [NewsEntity::class], version = 1)
abstract class NewsDb: RoomDatabase() {
    abstract fun dao(): NewsDao
    companion object {
        @Volatile private var I: NewsDb? = null
        fun get(c: Context) = I ?: synchronized(this) { I ?: Room.databaseBuilder(c, NewsDb::class.java, "news.db").build().also { I = it } }
    }
}
